File: Modulo04/Heranca-CSharp/Pessoa.cs
using System;

public class Pessoa {

    private string nome;
    private int idade;
    static int contador = 0;

    public Pessoa(string nome, int idade) {
        this.nome = nome;
        this.idade = idade;
        Pessoa.incrementa();
    }

    ~Pessoa() {
        Pessoa.decrementa();
    }

    public string getNome() {
        return this.nome;
    }

    public void setNome(string value) {
        this.nome = value;
    }

    public int getIdade() {
        return this.idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }
    
    public void imprime() {
        Console.WriteLine("Nome: {0}", this.nome);
        Console.WriteLine("Idade: {0}", this.idade);
    }
    
    public int compare(Pessoa p) {
        return idade - p.idade;
    }

    public static int getContador() {
        return Pessoa.contador;
    }

    private static void incrementa() {
        Pessoa.contador++;
    }

    private static void decrementa() {
        Pessoa.contador--;
    }
}

File: Modulo04/AlunoProfessor-Java/Pessoa.java

public class Pessoa {

    private String nome;
    private int idade;

    private static int contador = 0;

    public Pessoa(String nome, int idade) {
        this.nome = nome;
        this.idade = idade;
        incrementa();
        System.out.println("Construtor: " + this.nome);
    }

    @Override
    protected void finalize() throws Throwable {
        decrementa();
        System.out.println("Destrutor: " + this.nome);
    }
    
    public int getIdade() {
        return this.idade;
    }
    
    public void setIdade(int idade) {
        this.idade = idade;
    }
    
    public String getNome() {
        return this.nome;
    }
    
    public void setNome(String nome) {
        this.nome = nome;
    }
    
    public void imprime() {
        System.out.println("Nome: " + this.nome);
        System.out.println("Idade: " + this.idade);
    }
    
    public static int getContador() {
        return Pessoa.contador;
    }

    private static void decrementa() {
        Pessoa.contador -= 1;
    }

    private static void incrementa(){
        Pessoa.contador += 1;
    }
}

